Sources/Sentry/SentryProfiler.mm

```diff
@@ -254,7 +258,7 @@ - (SentryEnvelopeItem *)buildEnvelopeItemForTransaction:(SentryTransaction *)tra
     profile[@"device_is_emulator"] = @(isSimulatorBuild());
     profile[@"device_physical_memory_bytes"] =
         [@(NSProcessInfo.processInfo.physicalMemory) stringValue];
-    profile[@"environment"] = transaction.environment ?: @"production";
+    profile[@"environment"] = hub.scope.environmentString ?: hub.getClient.options.environment ?: kSentryDefaultEnvironment;
```

Comment:
Can we add a test for this? Search for `assertValidProfileData` to see where we verify the integrity of the profiling data in the current tests.